### DISCUSSION TITLE
automatisation du traitement des images

### DISCUSSION THREAD

**Discussion by nassertom** [marian-m12l/studio#220] [Ideas]
Bonjour,

l'automatisation des audios est faite via ffmpeg. pour les images c'est imagemagik (c'est aussi multi plateform...)

voici un exemple de script pour resize toutes les assets d'un pack :
`for file in /home/thomas/Bureau/resize/assets/*.png; do convert $file -resize 320x240^ -extent 320x240 $file; done`

vous en pensez quoi ?

**Comment by marian-m12l** [comment]
@nassertom Je viens de répondre sur le ticket #213. Côté audio, STUdio gère bien la conversion de format, et potentiellement de fréquence d'échantillonnage, mais ne modifie pas le "contenu" du fichier audio en lui-même (j'entends par là le volume, la durée, les blancs, etc.). De la même manière, pour les images on ne fait que convertir le format de fichier, avec potentiellement la génération d'une palette de couleurs sans quoi le nouveau firmware ne pourrait pas lire les bitmaps ; mais il n'y a pas de redimensionnement ou de "rognage" de l'image, qui reste de la responsabilité de l'utilisateur.
